src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,79 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException(System.lineSeparator() + "Can't create class with injector. "
+                    + "This class has not annotation "
+                    + "\"Componenet\". " + System.lineSeparator()
+                    + "Class name: " + clazz.getName());
+        }
+        Object clazzImplementationInstance = null;
+        Field[] declaredFields = clazz.getDeclaredFields();
+
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent((Inject.class))) {
+                Object fieldInstance = getInstance(field.getType());
+                clazzImplementationInstance = createNewInstance(clazz);
+                field.setAccessible(true);
+                try {
+                    field.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialize value. "
+                            + "Class: " + clazz.getName()
+                            + ". Field: " + field.getName());
```

Comment:
```suggestion
                            + ". Field: " + field.getName());
```
```suggestion
                            + ". Field: " + field.getName(), e);
```
 не забувай прокидувати помилку (перевір це в інших випадках)